File: Types/File.ts
interface Files {
	deleted: boolean;
	fileId: string;
	// ! not always present
	forId: string | null;
	// ! not always present
	hash: string | null;
	name: string;
	type: string;
	uploaded: boolean;
	uploadedAt: Date;
	uploadedBy: string;
}

export default Files;

export type { Files };

File: Structs/File.go
package structs

import (
	"time"

	"github.com/scylladb/gocqlx/v2/table"
)

var FilesTable = table.New(table.Metadata{
	Name:    "files",
	Columns: []string{"file_id", "for_id", "name", "type", "deleted", "uploaded", "uploaded_at", "uploaded_by", "hash", "int_tbl_ver"},
	PartKey: []string{"file_id"},
	SortKey: []string{},
})

type File struct {
	FileID     string    `db:"file_id"`
	ForID      *string   `db:"for_id"`
	Name       string    `db:"name"`
	Type       string    `db:"type"`
	Deleted    bool      `db:"deleted"`
	Uploaded   bool      `db:"uploaded"`
	UploadedAt time.Time `db:"uploaded_at"`
	UploadedBy string    `db:"uploaded_by"`
	Hash       *string   `db:"hash"`
	IntTblVer  int       `db:"int_tbl_ver"`
}
